src/routes/write-exercise/components/WriteExercise.tsx

```diff
@@ -41,14 +42,23 @@ const WriteExercise = ({ documents, route, navigation }: WriteExercisePropType):
     }))
   )
 
+  const isKeyboardShown = useKeyboard()
   const current = documentsWithResults[currentIndex]
   const nthRetry = current.numberOfTries ?? 0
   const needsToBeRepeated = nthRetry < numberOfMaxRetries && current.result !== SIMPLE_RESULTS.correct
 
   const tryLater = useCallback(() => {
-    setDocumentsWithResults(moveToEnd(documentsWithResults, currentIndex))
-    Keyboard.dismiss()
-  }, [documentsWithResults, currentIndex])
+    // ImageViewer is not resized correctly if keyboard is not dismissed before going to next document
+    if (isKeyboardShown) {
+      const onKeyboardHideSubscription = Keyboard.addListener('keyboardDidHide', () => {
```

Comment:
Why is this subscription needed?

Reply:
because we have to wait until the keyboard is hidden to go to the next word, so you have to listen to this event

Reply:
Oke but why? What would happen if we don't wait?

Reply:
check comment above :) ImageViewwer doesn't resize correctly

Reply:
oke, i see. thanks